### PR TITLE
sds011 communication

### PR DESCRIPTION
**Pull request by nyaapa**

### CODE REVIEW

File: sds011/sds011.cpp [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+	unsigned char year;
+	unsigned char month;
+	unsigned char day;
+	unsigned char id_le[2];
+	unsigned char crc;
+	unsigned char end;
+};
+
+struct data {
+	unsigned char pm25_le[2];
```
Comment (proowl) [inline sds011/sds011.cpp]: uint8_t?

File: sds011/sds011.cpp [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+		tcflush(fh, TCIOFLUSH);
+	} catch (...) {
+		close(fh);
+		throw;
+	}
+}
+
+sds011::~sds011() {
+	if (tcsetattr(fh, TCSANOW, &tty_back) < 0)
+		fmt::print(stderr, "Failed to reset tcsetattr: {}", strerror(errno));
```
Comment (proowl) [inline sds011/sds011.cpp:133]: exit(-1)?

File: sds011/sds011.cpp [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+	version& x = *reinterpret_cast<version*>(&response[3]);
+	unsigned char checksum = std::accumulate(&response[2], &response[8], 0u);
+
+	fmt::print(
+	    "Y: {}, M: {}, D: {}, ID: 0x{:x}, CRC={}\n", x.year, x.month, x.day, parse_le(x.id_le),
+	    x.crc == checksum && x.end == 0xab ? "OK" : "NOK");
+}
+
+void sds011::print_data() {
+	if (response[1] != 0xc0) {
```
Comment (proowl) [inline sds011/sds011.cpp]: 0xc0?

Comment (nyaapa) [reply sds011/sds011.cpp]: moved to constants

File: sds011/sds011.cpp (hunk 2 above)
Comment (nyaapa) [reply sds011/sds011.cpp:133]: don't think it's a critical one; new tty reader/writer should set it own attrs anyways

**Pull request merged**